challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java

```diff
@@ -0,0 +1,47 @@
+package challenges.stacksandqueues;
+
+public class PseudoQueue {
+    Stack stackOne;
+    Stack stackTwo;
+
+    public PseudoQueue(){
+        stackOne = new Stack();
+        stackTwo = new Stack();
+    }
+
+    public void push(Stack<Integer> topNode, int newValue) {
+        // push the data onto the stack
+        topNode.push(newValue);
+    }
+
+    public int pop(Stack<Integer> topNode){
+        if (topNode.isEmpty()) { // if stack is empty, throw an error
+            System.out.println("Stack Empty");
+            System.exit(0);
+        }
+        return topNode.pop(); //pop the value form the stack
+    }
+    public void enQueue(int value){
+//        push(stackOne, value);
+        stackOne.push(value);
+    }
+
+    public void deQueue(int value){
+       // int value;
+
+        // if both stacks are empty, throw an error
+        if(stackOne.isEmpty() && stackTwo.isEmpty()) {
+            System.out.println("Stack is empty");
+            System.exit(0);
+        }
+        // move elements from StackOne to stackTwo only if stack two is empty
```

Comment:
You are missing a bit of logic in the enqueue. You are checking below if stacktwo is empty, enqueue into stack one. But what if stackone is empty but stacktwo is not?

Reply:
Great point Matthew! It felt a little incomplete. thank you for the feedback